### PR TITLE
[fix bug 1195369] Redirect iOS users on /firefox/new/ to iOS product page.

### PR DESCRIPTION
**Pull request by jpetto**
This PR must be merged after #3219, but can be code reviewed now.

Currently up on demo3, and can be tested with a (temporary) URL hack, e.g.

http://www-demo3.allizom.org/firefox/new/?referrer=http://www.google.com

### CODE REVIEW

**Comment by jpetto** [comment]
@schalkneethling or @sgarrity r?

**Comment by alexgibson** [comment]
Would be good to add a test for `getDomainFromURL` that correctly handles a non valid URL, just for good measure?

File: media/js/firefox/new-ios-redirect-helper.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+if (typeof window.Mozilla === 'undefined') {
+    window.Mozilla = {};
+}
+
+(function(Mozilla) {
+    'use strict';
+
+    Mozilla.FirefoxNewIosRedirectHelper = {
+        getDomainFromURL: function(url) {
+            var matches = url.match(/^https?\:\/\/([^\/?#]+)(?:[\/?#]|$)/i);
```
Comment (alexgibson) [inline media/js/firefox/new-ios-redirect-helper.js]: Regex isn't one of my strong points, but out of curiosity what is the non-capturing group `(?:[\/?#]|$)` needed for? It doesn't seem to be an issue, but the tests seem to pass both with and without it.

**Comment by alexgibson** [comment]
Code looks good, and testing on demo3 using an iOS device it seems to be working as expected. One question/suggestion but otherwise it's an r+ from me (optional wc).

Before this merges, do we need to do any sort of GA verification on demo3?

File: media/js/firefox/new-ios-redirect-helper.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                domain = domain.replace('www.', '');
+            }
+
+            return domain;
+        },
+        isSearchEngine: function(domain) {
+            var result = false;
+
+            // only concerned with these top 4 for now
+            if ((/bing\.|duckduckgo\.|google\.|yahoo\./i).test(domain)) {
```
Comment (alexgibson) [inline media/js/firefox/new-ios-redirect-helper.js]: Should we be checking specifically for `search.yahoo` here, as they have several web properties?

File: media/js/firefox/new-ios-redirect-helper.js (hunk 1 above)
Comment (jpetto) [inline media/js/firefox/new-ios-redirect-helper.js]: Clearly not my strongest point, either. :grin:

You're correct - it isn't needed. Was trying to be too fancy by accounting for trailing querystring or hash, but there doesn't seem to be a need to care about anything beyond the domain itself. Removing.

File: media/js/firefox/new-ios-redirect-helper.js (hunk 2 above)
Comment (jpetto) [inline media/js/firefox/new-ios-redirect-helper.js]: Good call. Confirmed with Gareth - will update to `search.yahoo.`.

**Comment by jpetto** [comment]
@alexgibson Updated!

Gareth did test on demo3 to make sure the `utm_` params were coming through as expected. I don't *think* we need additional testing.

**Comment by alexgibson** [comment]
Changes look good, tests pass r+ from me to merge when ready :+1:

**Comment by jpetto** [comment]
Added a waffle switch to the JS block that performs the redirect.

**Comment by jpetto** [comment]
Waffle switch name is `firefox-new-ios-redirect`.

**Comment by craigcook** [comment]
Waffle working as expected. r++! :bamboo:

**Pull request merged**